drivers/CoreRFIDReader/tests/CoreRFIDReader_test.cpp

```diff
@@ -0,0 +1,264 @@
+
+	mock::EventQueue event_queue {};
+	mock::CoreBufferedSerial mockBufferedSerial;
+	CoreRFIDReader reader;
+
+	bool _callbackCalled {false};
+
+	static void func1() {};
+	static void func2(rfid::Tag tag) {};
+	std::function<void()> callbackDetected			= func1;
+	std::function<void(rfid::Tag)> callbackReadable = func2;
+	std::function<void()> callbackSigio {};
+
+	void isCallbackCalled() { _callbackCalled = true; };
+
+	void setFakeCallback()
+	{
+		_callbackCalled = false;
+		auto callback	= [this]() { this->isCallbackCalled(); };
+		reader.registerOnTagDetectedCallback(callback);
+	}
+
+	void sendSetProtocol()
+	{
+		const auto expected_values = ElementsAre(rfid::command::set_protocol::id, rfid::command::set_protocol::length,
+												 rfid::protocol::iso14443A.id, rfid::settings::default_rx_tx_speed);
+		EXPECT_CALL(mockBufferedSerial, write).With(Args<0, 1>(expected_values));
+	}
+
+	void sendSetGainAndModulation()
+	{
+		const auto expected_values = ElementsAre(
+			rfid::command::set_gain_and_modulation::id, rfid::command::set_gain_and_modulation::length,
+			rfid::settings::arc_b, rfid::settings::flag_increment, rfid::settings::acr_b_index_for_gain_and_modulation,
+			rfid::protocol::iso14443A.gain_modulation_values());
+		EXPECT_CALL(mockBufferedSerial, write).With(Args<0, 1>(expected_values));
+	}
+
+	template <size_t size>
+	void receiveRFIDReaderAnswer(const std::array<uint8_t, size> &returned_values)
+	{
+		EXPECT_CALL(mockBufferedSerial, readable).WillOnce(Return(true));
+		EXPECT_CALL(mockBufferedSerial, read)
+			.WillOnce(DoAll(SetArrayArgument<0>(begin(returned_values), begin(returned_values) + size), Return(size)));
+	}
+};
+
+TEST_F(CoreRFIDReaderTest, initialization)
+{
+	EXPECT_NE(&reader, nullptr);
+}
+
+TEST_F(CoreRFIDReaderTest, init)
+{
+	EXPECT_CALL(mockBufferedSerial, sigio).Times(1);
+
+	reader.init();
+}
+
+TEST_F(CoreRFIDReaderTest, registerCallbacks)
+{
+	reader.registerOnTagDetectedCallback(callbackDetected);
+	reader.registerOnTagReadableCallback(callbackReadable);
+}
+
+TEST_F(CoreRFIDReaderTest, setModeTagDetection)
+{
+	const auto expected_values =
+		ElementsAre(rfid::command::frame::set_tag_detection_mode[0], rfid::command::frame::set_tag_detection_mode[1],
+					rfid::command::frame::set_tag_detection_mode[2], rfid::command::frame::set_tag_detection_mode[3],
+					rfid::command::frame::set_tag_detection_mode[4], rfid::command::frame::set_tag_detection_mode[5],
+					rfid::command::frame::set_tag_detection_mode[6], rfid::command::frame::set_tag_detection_mode[7],
+					rfid::command::frame::set_tag_detection_mode[8], rfid::command::frame::set_tag_detection_mode[9],
+					rfid::command::frame::set_tag_detection_mode[10], rfid::command::frame::set_tag_detection_mode[11],
+					rfid::command::frame::set_tag_detection_mode[12], rfid::command::frame::set_tag_detection_mode[13],
+					rfid::command::frame::set_tag_detection_mode[14], rfid::command::frame::set_tag_detection_mode[15]);
```

Comment:
Comme il n'y pas de valeur brute à préciser et que tout est déjà dans un array spécifique

```suggestion
	const auto expected_values = testing::ElementsAreArray(rfid::command::frame::set_tag_detection_mode);
```